packages/@ember/-internals/routing/lib/ext/controller.ts

```diff
@@ -9,6 +10,15 @@ import { deprecateTransitionMethods, prefixRouteNameArg } from '../utils';
 ControllerMixin.reopen({
   concatenatedProperties: ['queryParams'],
 
+  init() {
+    this._super(...arguments);
+    let owner = getOwner(this);
+    if (owner) {
+      this.namespace = owner.lookup('application:main');
+      this.target = owner.lookup('router:main');
+    }
```

Comment:
Do we want to ensure every test has the right owner setup so we don't have to guard?

Reply:
e.g. see `controller_test.js` where I did this